### PR TITLE
docs: Cloudflare guide update with example

### PR DESCRIPTION
**Pull request by rmarscher**
Moving a code example for Cloudflare to Waku so we don't need to reference my public gist and it's easier for everyone to test.

### CODE REVIEW

File: examples/26_cloudflare/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    "build": "waku build --with-cloudflare",
+    "start": "waku start"
+  },
+  "dependencies": {
+    "hono": "^4.6.3",
+    "react": "19.0.0-rc-2d16326d-20240930",
+    "react-dom": "19.0.0-rc-2d16326d-20240930",
+    "react-server-dom-webpack": "19.0.0-rc-2d16326d-20240930",
+    "waku": "0.21.3",
+    "wrangler": "^3.80.0"
```
Comment (dai-shi) [inline examples/26_cloudflare/package.json:18]: we pin versions.
```suggestion
    "hono": "4.6.3",
    "react": "19.0.0-rc-2d16326d-20240930",
    "react-dom": "19.0.0-rc-2d16326d-20240930",
    "react-server-dom-webpack": "19.0.0-rc-2d16326d-20240930",
    "waku": "0.21.3",
    "wrangler": "3.80.0"
```

**Review by dai-shi — COMMENTED** [review]
thanks for working on this!

**Review by dai-shi — CHANGES_REQUESTED** [review]
rename 26_cloudflare to 44_cloudflare and add entries in package.json "scripts".

**Comment by rmarscher** [comment]
Thank you. I will make these updates after I fix #944. Switching to draft for now.

**Comment by rmarscher** [comment]
I think this should be ready once #989 is merged.

File: package.json [hunk 2]
```diff
@@ -53,6 +53,7 @@
     "examples:prd:41_path-alias": "NAME=41_path-alias pnpm run examples:prd",
     "examples:prd:42_react-tweet": "NAME=42_react-tweet pnpm run examples:prd",
     "examples:prd:43_weave-render": "NAME=43_weave-render pnpm run examples:prd",
+    "examples:prd:44_cloudflare": "NAME=44_cloudflare pnpm run examples:prd",
```
Comment (dai-shi) [inline package.json:56]: We also have `examples:dev:*` entry.

File: examples/44_cloudflare/package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    "waku": "0.21.5",
+    "wrangler": "3.84.1"
+  },
+  "devDependencies": {
+    "@types/react": "npm:types-react@rc",
+    "@types/react-dom": "npm:types-react-dom@rc",
+    "typescript": "5.6.2"
+  },
+  "prettier": {
+    "singleQuote": true
```
Comment (dai-shi) [inline examples/44_cloudflare/package.json:26]:
```suggestion
```

**Review by dai-shi — COMMENTED** [review]
LGTM otherwise.

File: examples/44_cloudflare/package.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    "dev": "waku dev",
+    "build": "waku build --with-cloudflare",
+    "start": "waku start"
+  },
+  "dependencies": {
+    "hono": "4.6.5",
+    "react": "19.0.0-rc-bf7e210c-20241017",
+    "react-dom": "19.0.0-rc-bf7e210c-20241017",
+    "react-server-dom-webpack": "19.0.0-rc-bf7e210c-20241017",
+    "waku": "0.21.5",
```
Comment (dai-shi) [inline examples/44_cloudflare/package.json:17]: This is updated. Check the main branch.

**Review by dai-shi — CHANGES_REQUESTED** [review]

File: examples/44_cloudflare/package.json (hunk 4 above)
Comment (rmarscher) [reply examples/44_cloudflare/package.json]: Prettier error... one moment...

Comment (rmarscher) [reply examples/44_cloudflare/package.json]: New error `error TS2688: Cannot find type definition file for './.wrangler/types/runtime.d.ts'.`  🙄  I'll look for a workaround.

**Comment by rmarscher** [comment]
It looks like running the example for E2E is hitting a 401 Unauthorized error. I guess running the miniflare dev server communicates with Cloudflare's API and is expecting CLOUDFLARE_API_TOKEN and CLOUDFLARE_ACCOUNT_ID to be set in the environment.

**Comment by rmarscher** [comment]
I tried a couple workarounds. I need to set up a copy of the CI environment so I can figure out how to workaround the authorization error. I think adding actual CLOUDFLARE_API_TOKEN and CLOUDFLARE_ACCOUNT_ID secrets to the GitHub CI environment would also fix it.

**Comment by dai-shi** [comment]
I don't think it's a CI issue, but I can't run the test without those envvars, right? Doesn't it need a mock auth server?

**Comment by rmarscher** [comment]
I think the authorization error that I saw in the CI logs was unrelated. I had an error from updating to the latest minimal client. I'm guessing that it was working locally for me because I had cached vite assets or something like that.

**Review by dai-shi — APPROVED** [review]
LGTM

**Pull request merged**